vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java

```diff
@@ -370,8 +370,10 @@ private void resetContent() {
 
     private void onTargetAttach(UI ui) {
         ui.getInternals().addComponentDependencies(ContextMenu.class);
-        ui.getPage().executeJs(
-                "window.Vaadin.Flow.contextMenuConnector.init($0)", target);
+        ui.beforeClientResponse(target, context -> {
```

Comment:
I'm not sure how big of a deal this is for such a corner case, but each time the attach-detach combo is executed, it registers a new execution, which is not actually cancelled, but postponed until the target remains attached. So in the test view, spamming the `attach-detach`-button can bloat the server memory with these executions. Once the target is actually attached, it will flush all of those.

That could be improved by always running the execution, but checking if the target is actually attached at that point. To always run instead of postpone the execution, you can use `ui` instead of `target` as the first parameter.
```java
ui.beforeClientResponse(ui, context -> {
    if (target.getUI().isPresent()) {
        ui.getPage().executeJs(
                "window.Vaadin.Flow.contextMenuConnector.init($0)",
                target);
    }
});
```